kratos/utilities/volume_inside_voxel_utility.h

```diff
@@ -0,0 +1,194 @@
+ * @ingroup KratosCore
+ * @brief Utilities to compute the real volume inside a voxel
+ * @details This class provides static methods to compute (using different approximations) the portion of a 
+ * voxel that is actually filled with volume, according to the known triangle elements that intersect the voxel
+ * @author Ariadna Cortés
+ */
+class VolumeInsideVoxelUtility
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Node<3> NodeType;
+    typedef Node<3>::Pointer NodePtrType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef GeometryType::Pointer GeometryPtrType;
+    typedef GeometryType::GeometriesArrayType GeometryArrayType;
+    typedef GeometryType::PointsArrayType PointsArrayType;
+
+
+    /// Pointer definition of VoxelInsideVolume
+    KRATOS_CLASS_POINTER_DEFINITION( VolumeInsideVoxelUtility );
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    VolumeInsideVoxelUtility(){}
+
+    /// Destructor
+    virtual ~VolumeInsideVoxelUtility(){}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Aproximates the actual volume inside the voxel 
+     * @param rVoxel references to the voxel whose actual volume will be approximated
+     * @return Approximated volume 
+     * @note This approximation assigns a fraction of volume (1/8) to each node of the
+     * voxel, and counts it as volume if the node is inside the object (NodeDistance > 0)
+     * This operation is VERY cheap
+     */  
+    static double NodesApproximation(const GeometryType& rVoxel);
+
+    /*This method is completly useless since it does the same calculation as the previous 
+    one but in a different way. Helps to illustrate use of Edges */
+    static double EdgesApproximation(const GeometryType& rVoxel);
+    
+    /**
+     * @brief Aproximates the actual volume inside the voxel 
+     * @param rVoxel references to the voxel whose actual volume will be approximated
+     * @param rTriangles references to the triangles which intersect the voxel at some edge.
+     * @return Approximated volume 
+     * @note This approximation finds the portion of each edge that is part of the volume (using
+     * intersection points with triangles of the mesh).
+     */  
+    static double EdgesPortionApproximation(const GeometryType& rVoxel, const GeometryArrayType& rTriangles);
+
+    /**
+     * @brief Aproximates the actual area inside a quadrilateral with 90º angles
+     * @param rFace references to the quadrilateral3D4 whose actual area will be approximated
+     * @param rTriangles references to the triangles which may intersect the quadrilateral at some edge.
+     * @return Approximated area 
+     * @note this method is cheaper than HexaVolume2D, but won't work correctly with != 90º angles
+     */  
+    static double VoxelVolume2D(const GeometryType& rFace, const GeometryArrayType& rTriangles);
```

Comment:
This naming is way misleading... Also the documentation. It is not clear for future users what this method does.